src/rosdep2/main.py

```diff
@@ -391,6 +392,18 @@ def _rosdep_main(args):
         check_for_sources_list_init(options.sources_cache_dir)
     elif command not in ['fix-permissions']:
         setup_proxy_opener()
+
+    if 'ROS_PYTHON_VERSION' not in os.environ and options.ros_distro:
```

Comment:
Shouldn't `options.ros_distro` take precedence over the environment variable?

Reply:
`options.ros_distro` overriding `ROS_PYTHON_VERSION` might be a problem for Arch Linux users. They seem to be using Python 3: https://aur.archlinux.org/packages/ros-melodic-catkin/ with melodic. If `options.ros_distro` takes precedence then with #709 pip dependencies would be installed with Python 2 using Melodic on Arch.

Reply:
See https://github.com/ros-infrastructure/rosdep/pull/708#discussion_r331204564 why on other platforms the rosdistro configured value must have precedence.

Reply:
@wjwwood I think this is resolved by dd4bf62 (explanation in https://github.com/ros-infrastructure/rosdep/pull/708#discussion_r331204564). Would you mind having another look at this PR?